### PR TITLE
User Account Activation

### PR DESCRIPTION
**Pull request by YongRongXin**

### CODE REVIEW

**Comment by YongRongXin** [comment]
ready

File: app/mailers/application_mailer.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
 class ApplicationMailer < ActionMailer::Base
-  default from: "from@example.com"
+  default from: "noreply@onlinecourses.com"
```
Comment (saobv-2223) [inline app/mailers/application_mailer.rb]: using ENV for it

File: app/mailers/user_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    @user = user
+    mail to: user.email, subject: t("mailers.user.subject")
+  end
+
+  def password_reset
+    @greeting = "Hi"
+
+    mail to: "to@example.org"
+  end
+end
```
Comment (saobv-2223) [inline app/mailers/user_mailer.rb]:
```suggestion
end

```

File: app/models/user.rb [hunk 3]
```diff
@@ -35,17 +36,31 @@ def remember
     update_attribute :remember_digest, User.digest(remember_token)
   end
 
-  def authenticated? remember_token
-    return false if remember_digest.nil?
-    BCrypt::Password.new(remember_digest).is_password? remember_token
+  def authenticated? attribute, token
+    digest = send "#{attribute}_digest"
+    return false if digest.nil?
+    BCrypt::Password.new(digest).is_password? token
```
Comment (saobv-2223) [inline app/models/user.rb]:
```suggestion
    return false unless digest
    
    BCrypt::Password.new(digest).is_password? token
```

File: app/mailers/user_mailer.rb (hunk 2 above)
Comment (YongRongXin) [reply app/mailers/user_mailer.rb]: done

File: app/models/user.rb (hunk 3 above)
Comment (YongRongXin) [reply app/models/user.rb]: done

**Comment by YongRongXin** [comment]
ready

File: app/mailers/application_mailer.rb (hunk 1 above)
Comment (YongRongXin) [reply app/mailers/application_mailer.rb]: done

**Review by saobv-2223 — COMMENTED** [review]
merge

**Pull request merged**